src/components/datepicker/datePicker.js

```diff
@@ -214,7 +216,10 @@
 
       if (self.dateUtil.isValidDate(parsedDate) && self.dateLocale.isDateComplete(inputString)) {
         self.ngModelCtrl.$setViewValue(parsedDate);
+        self.inputContainer.classList.remove(INVALID_CLASS);
         self.$scope.$apply();
+      } else if (inputString) {
+        self.inputContainer.classList.add(INVALID_CLASS);
```

Comment:
If you go from the invalid state to an empty string, in invalid state doesn't go away; the invalid class should always go away if the input is empty.